src/ekat/kokkos/ekat_kokkos_utils.hpp

```diff
@@ -521,21 +501,6 @@ class TeamUtils<ValueType,Kokkos::Cuda> : public TeamUtilsCommonBase<ValueType,K
     }
   }
 
-  TeamUtils& operator= (const TeamUtils& src)
```

Comment:
If you are ok with the default op=, then I would suggest making it clear, by declaring op= defaulted.

Reply:
Ok yeah, I was about to ask that question. What is the difference in eliminating it vs. adding the default declaration?

Reply:
Morally, none. The compiler auto-generates the default op= anyways. However, imho there are a couple of benefits adding an explicitly defaulted one. The most important one is that the reader _knows_ you are ok with the default impl. The second is more personal: if down the road your class becomes non-copyable (e.g., it contains a member that is non-copyable), you would get a slightly better error message (at least with gcc):

without defaulted op=:
```
error: use of deleted function ‘B& B::operator=(const B&)’
   32 |   b2 = b1;
      |        ^~
note: ‘B& B::operator=(const B&)’ is implicitly deleted because the default definition would be ill-formed:
   23 | struct B {
      |        ^
```
with defaulted op=:
```
error: use of deleted function ‘B& B::operator=(const B&)’
   32 |   b2 = b1;
      |        ^~
note: ‘B& B::operator=(const B&)’ is implicitly deleted because the default definition would be ill-formed:
   24 |   B& operator= (const B&) = default;
      |      ^~~~~~~~
```
Not a big difference, but to me, it's always clearer when the compiler points to a problematic method (like in the second case) rather than to the class name (like in the first case).